sklearn/calibration.py

```diff
@@ -25,6 +27,357 @@
+        if (not hasattr(self.base_estimator, 'decision_function') and
+                not hasattr(self.base_estimator, 'predict_proba')):
+            raise TypeError('The base_estimator needs to implement either a '
+                            'decision_function or a predict_proba method')
+
+        if self.strategy not in ('roc', 'f_beta', 'max_tpr', 'max_tnr'):
+            raise ValueError('strategy can either be "roc" or "max_tpr" or '
+                             '"max_tnr. Got {} instead'.format(self.strategy))
+
+        if self.method not in (None, 'decision_function', 'predict_proba'):
+            raise ValueError('method param can either be "decision_function" '
+                             'or "predict_proba" or None. '
+                             'Got {} instead'.format(self.method))
+
+        if self.strategy == 'max_tpr' or self.strategy == 'max_tnr':
+            if (not self.threshold or not
+                    isinstance(self.threshold, (int, float))
+                    or not self.threshold >= 0 or not self.threshold <= 1):
+                raise ValueError('parameter threshold must be a number in'
+                                 '[0, 1]. '
+                                 'Got {} instead'.format(self.threshold))
+
+        if self.strategy == 'f_beta':
+            if not self.beta or not isinstance(self.beta, (int, float)):
+                raise ValueError('parameter beta must be a real number.'
+                                 'Got {} instead'.format(type(self.beta)))
+
+        X, y = check_X_y(X, y)
+
+        y_type = type_of_target(y)
+        if y_type != 'binary':
+            raise ValueError('Expected target of binary type. Got {}'.format(
+                y_type))
+
+        self.label_encoder_ = LabelEncoder().fit(y)
+
+        y = self.label_encoder_.transform(y)
+        self.pos_label = self.label_encoder_.transform([self.pos_label])[0]
+
+        if self.cv == 'prefit':
+            self.decision_threshold_ = _CutoffClassifier(
+                self.base_estimator, self.strategy, self.method, self.beta,
+                self.threshold, self.pos_label
+            ).fit(X, y).decision_threshold_
+            self.std_ = .0
+        else:
+            cv = check_cv(self.cv, y, classifier=True)
+            decision_thresholds = []
+
+            for train, test in cv.split(X, y):
+                estimator = clone(self.base_estimator).fit(X[train], y[train])
+                decision_thresholds.append(
+                    _CutoffClassifier(estimator, self.strategy, self.method,
+                                      self.beta, self.threshold,
+                                      self.pos_label).fit(
+                        X[test], y[test]
+                    ).decision_threshold_
+                )
+            self.decision_threshold_ = np.mean(decision_thresholds)
+            self.std_ = np.std(decision_thresholds)
+
+            self.base_estimator.fit(X, y)
+        return self
```

Comment:
As a classifier, CutoffClassifierCV must provide a classes_ attribute after fitting

Reply:
Hey Although I have added this, I am thinking, shouldn't this be checked on the base estimator integration tests? (the existence of classes_ after fit)

Reply:
It should be, but at the moment, out tests do not attempt to run on meta-estimators. :(

Actually, it would be a good idea to force checking this with the full common test suite. Add something like:

```
from sklearn.utils.estimator_checks import check_estimator

def test_cutoff_classifier_common_tests:
     check_estimator(CutoffClassifierCV(LogisticRegression()))
```

Reply:
Hm I guess all our tests might be multi-class? Can you check? If there's a significant portion of tests that are not multi-class, just manually skip the multi-class ones.